### ISSUE TITLE
Watch never restarts the server after multiple watch events (v4.2.0)

### ISSUE THREAD

**Issue by bcomnes** [fastify/fastify-cli#518]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.2.0

### CLI version

4.2.0

### Node.js version

18.4.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

12.4

### Description

After updating to fastify-cli 4.2.0 from 4.1.1, it seems as though modifying watched files can cause the fastify-cli server to stop the server and never restart. Killing the process after this happens, the following error is observed sometimes:

```console
% Error [ERR_IPC_CHANNEL_CLOSED]: Channel closed
[fastify-cli] app crashed - waiting for file changes before starting...
[1656790027711] INFO (42069 on MacBook-Pro.bec.local): Server listening at http://0.0.0.0:3000
```

A samble of verbose logging prior to the server not responding/restarting looks like this:

```
[fastify-cli] watch - 'addDir' occurred on 'public/hooks'
[fastify-cli] watch - 'addDir' occurred on 'public/lib'
[fastify-cli] watch - 'addDir' occurred on 'public/login'
[fastify-cli] watch - 'addDir' occurred on 'public/logout'
[fastify-cli] watch - 'addDir' occurred on 'public/register'
[fastify-cli] watch - 'addDir' occurred on 'public/resources'
[fastify-cli] watch - 'addDir' occurred on 'public/tags'
```

After these messages print, the server is no longe accepting requests. 

### Steps to Reproduce

Start a fastify-cli server in watch mode running fastify-cli 4.2.0 and then trigger some file changes. A `mkdir -p` works well. The server restarts if there are only one or two file watch actions at a time. It will not restart if there are 3-4 actions at a time. Downgrading the CLI to 4.1.1 restore a functional watch mode. 

[Sample repo](https://github.com/ballpit/fastify-cli-bug)

### Expected Behavior

Server should eventually accept requests after files change in watch mode.

**Comment by bcomnes** [comment]
Caused by this patch: https://github.com/fastify/fastify-cli/pull/511

**Comment by bcomnes** [comment]
Closing in favor of https://github.com/fastify/fastify-cli/pull/515 and https://github.com/fastify/fastify-cli/issues/513
